### PR TITLE
DE46284 - Fix date localization by replacing `this.locale` with `this.language`

### PR DESCRIPTION
**Pull request by jameswklassen**
## Issue 🐛 

Dates weren't being localized because:
- `this.locale` was being used to localize dates.
- `this.locale` was `undefined` (so date localization failed and used english as a fallback)

## Solution ✅ 
- Use `this.language` (which we get for free) to localize dates

## Rally
https://rally1.rallydev.com/#/378960033600d/custom/382879628416?detail=%2Fdefect%2F618698567095

### CODE REVIEW

File: components/d2l-all-assessments-list.js [hunk 1]
```diff
@@ -126,7 +126,7 @@ Polymer({
 		var dateStringPrefix;
 		var calendarDateDiff = this.getDateDiffInCalendarDays(activity[0].dueDate || activity[0].endDate);
 
-		dateString = new Intl.DateTimeFormat(this.locale, { weekday: 'long', month: 'long', day: 'numeric' }).format(new Date(activity[0].dueDate || activity[0].endDate));
+		dateString = new Intl.DateTimeFormat('this.language', { weekday: 'long', month: 'long', day: 'numeric' }).format(new Date(activity[0].dueDate || activity[0].endDate));
```
Comment (jstefaniuk-d2l) [inline components/d2l-all-assessments-list.js:129]:
```suggestion
		dateString = new Intl.DateTimeFormat(this.language, { weekday: 'long', month: 'long', day: 'numeric' }).format(new Date(activity[0].dueDate || activity[0].endDate));
```

Comment (jameswklassen) [reply components/d2l-all-assessments-list.js]: Oh, that's weird. Not sure how that happened 😅

**Review by jstefaniuk-d2l — APPROVED** [review]
code LGTM

File: wct.conf.json [hunk 2]
```diff
@@ -12,23 +12,23 @@
       "browsers": [
         {
           "browserName": "chrome",
-          "platform": "OS X 10.13",
-          "version": ""
+          "platform": "macOS 11.00",
```
Comment (jadasc) [inline wct.conf.json:15]: How did you know to switch the platform? And which one to switch it to?

Comment (jameswklassen) [reply wct.conf.json:15]: Oh, I was just experimenting with our test platforms to see if I could get CI to pass.

I decided to just open a fresh PR to fix CI though, so I'll remove the file change here.

**Comment by marcsoiferman** [comment]
Should you revert the changes in `wct.conf.json` in this PR if it's being done in #191?

**Comment by jameswklassen** [comment]
> Should you revert the changes in `wct.conf.json` in this PR if it's being done in #191?

Yeah, I was trying to fix CI in this PR, but I ended up just opening a new one to have a fresh slate.

**Review by marcsoiferman — APPROVED** [review]
Code LGTM

**Pull request merged**